modules/weko-schema-ui/weko_schema_ui/serializers/WekoBibTexSerializer.py

```diff
@@ -304,135 +481,234 @@ def get_jpcoar_data(pid, record):
+            else:
+                creator[BibTexFields.YOMI.value].append(
+                    element.text)
+
+        def process_url():
+            identifier_type = element.get(xml_ns + 'identifierType')
+            if identifier_type and identifier_type.lower in self.__lst_identifier_type:
+                lst_identifier_type_data[
+                    identifier_type.lower].append(element.text)
 
-        """
-        # Initialization
         data = {}
         page_start = ''
         page_end = ''
-        xml_ns = '{' + self.ns['xml'] + '}'
-
-        # Create book record
-        for field in bibtex_cols_all.keys():
-            elements = root.findall('.//' + bibtex_cols_all[field], self.ns)
+        title = ''
+        xml_ns = '{' + self.__ns['xml'] + '}'
+        creator = {BibTexFields.AUTHOR.value: [],
+                   BibTexFields.YOMI.value: []}
+        lst_identifier_type_data = {}
+        dois = []
+        all_fields = self.__combine_all_fields(bibtex_type)
+
+        for i in self.__lst_identifier_type:
+            lst_identifier_type_data[i] = []
+
+        for field in all_fields:
+            elements = root.findall(
+                self.__find_pattern.format(self.__fields_mapping[field]), self.__ns)
             if len(elements) != 0:
-
                 value = ''
                 dates = []
                 for element in elements:
-                    if field == 'date' and (element.get('dateType') is not None
-                                            and element.get('dateType').lower() == 'issued'):
-                        dates.append(element.text)
-                        continue
-                    elif field == 'type' and (element.get('descriptionType') is None
-                                              or element.get('descriptionType').lower() != 'other'):
-                        continue
-                    elif field == 'author' and (element.get(xml_ns + 'lang') is None
-                                                or element.get(xml_ns + 'lang').lower() != 'en'):
-                        continue
-
+                    if field == BibTexFields.YEAR or \
+                            field == BibTexFields.MONTH:
+                        process_by_att('dateType', 'issued', dates)
+                    elif field == BibTexFields.AUTHOR:
+                        process_author()
+                    elif field == BibTexFields.DOI:
+                        process_by_att(xml_ns + 'identifierType', 'doi', dois)
+                    elif field == BibTexFields.URL:
+                        process_url()
+                    elif field == BibTexFields.TITLE and title == '':
+                        # Get only one title at all
+                        title = element.text
                     if value != '':
-                        value += ' and ' if field == 'author' else ', '
+                        value += ' and ' if field == BibTexFields.AUTHOR else ', '
```

Comment:
' and ' is used many times, so create a variable to store ' and ' and reuse it.
Please similar check other string.